src/lib/Embed.hs

```diff
@@ -752,7 +760,7 @@ indexToIntE ty idx = case ty of
     foldM iadd (IntVal 0) scaled
   VariantTy types -> do
     sizes <- traverse indexSetSizeE types
-    (offsets, _) <- traverseFunM
+    (offsets, _) <- scanM
       (\sz prev -> do {v <- iadd sz prev; return (prev, v)}) sizes (IntVal 0)
```

Comment:
Oh a nice trick that might simplify the body of the lambda and let you avoid folding the line would be this:
```hs
\sz prev -> (prev,) <$> iadd sz prev
```